### DISCUSSION TITLE
UX thoughts

### DISCUSSION THREAD

**Discussion by Pulau-Komodo** [ferraridamiano/ConverterNOW#133] [Ideas]
I wasn't sure whether to make this an issue or a discussion, but I figured since it's sort of a subjective thing, I would put it here.

My experience is only with smartphone.

I appreciate the simplicity of the UX of entering a number as any unit and converting to all other units. That seems much more practical than Unit Converter Ultimate's approach of choosing two units. In Converter NOW, the user basically chooses the first unit when he selects an input field, and the second unit with his eyes. That's good.

But each unit seems to take much vertical space, increasing the chance that the user has to scroll to find the unit he wants. This could easily be condensed to further reduce user interactions.

A one-tap button to copy to clipboard would facilitate using the conversion in conversations, notes, and other applications. Theoretically this would also facilitate copying inside this application to do multi-dimensional conversions in multiple steps, but there are probably better ways to achieve that.

The ability to rearrange units is good, but I think it should be accessed from the screen for a specific property. I think it is likely that a user will be looking at the unit list when he decides to rearrange it, and the current approach requires him to navigate to where he already is.

**Comment by ferraridamiano** [comment]
Hi @Pulau-Komodo and thank you for this discussion!

> But each unit seems to take much vertical space, increasing the chance that the user has to scroll to find the unit he wants. This could easily be condensed to further reduce user interactions.

At the moment, the app adapts to the screen size, hence if you are using a tablet it will show a grid with 3-4 columns. If we double the columns also on smartphone I think that the user has to move the eyes left to right and back and scroll down in the meanwhile. While at the moment, (with one column on smartphone) the user has to keep the eye on the same spot and just scroll down. Tell me what do you think about this.

> A one-tap button to copy to clipboard would facilitate using the conversion in conversations, notes, and other applications.

I 100% agree on this and there is alreay an [open issue](https://github.com/ferraridamiano/ConverterNOW/issues/119). Some implementation ideas are missing, so please tell me your suggestion in that issue! We really need it!

> The ability to rearrange units is good, but I think it should be accessed from the screen for a specific property. I think it is likely that a user will be looking at the unit list when he decides to rearrange it, and the current approach requires him to navigate to where he already is.

This was the original design. It's changed in the 3.0 update. I moved it to the settings because I thought that it was a feature that was seldom used by the user and so it should not immediatly visible. But if there are any others opinion I'm open to move it back!

**Reply by Pulau-Komodo** [reply]
> At the moment, the app adapts to the screen size, hence if you are using a tablet it will show a grid with 3-4 columns. If we double the columns also on smartphone I think that the user has to move the eyes left to right and back and scroll down in the meanwhile. While at the moment, (with one column on smartphone) the user has to keep the eye on the same spot and just scroll down. Tell me what do you think about this.

I agree that columns have that downside. It probably would not bother me personally, but the application is not just for me! It seems like much vertical space could be saved without going to columns, though. There is a lot of padding between the elements, and probably more vertical space than needed inside each element. The unit could be to the left of the number, for example. I would like to hear more opinions on this.

> I 100% agree on this and there is alreay an [open issue](https://github.com/ferraridamiano/ConverterNOW/issues/119). Some implementation ideas are missing, so please tell me your suggestion in that issue! We really need it!

Your example looks good to me. The button could be to the left of the unit symbol if we are trying to save vertical space.

> This was the original design. It's changed in the 3.0 update. I moved it to the settings because I thought that it was a feature that was seldom used by the user and so it should not immediatly visible. But if there are any others opinion I'm open to move it back!

I agree that it is likely seldom used. But maybe something could be done anyway. I am not very familiar with Android's design language. Is some overflow menu in a top corner too much in-your-face? Is that how it used to be?

**Reply by ferraridamiano** [reply]
> The unit could be to the left of the number, for example

I tried many times but then the app loose too much color, become just white/black. It became uglier.

> Is some overflow menu in a top corner too much in-your-face? Is that how it used to be?

In pre 3.0 versions it was there. But I think that it also lacks in consistency:
- Where do you go if you want to reoder the properties? To the settings
- Where do you go if you want to reoder the units? ~To the relative property page~ To the settings

Since the user seldom reoder the units, the user need to discover that option when she/he need it. I think it is more discoverable in the settings.

There is also another problem of the reoder button inside the relative property page with large screens. Take a look at the app on desktop, where would you move that button?
![Screenshot_20220302_204250](https://user-images.githubusercontent.com/34270884/156437546-337acf14-48b4-4bbe-8825-08be90a337df.png)

**Reply by ferraridamiano** [reply]
Hi, what about this layout?
![new_1](https://user-images.githubusercontent.com/34270884/188701803-0e9857ad-7f43-46d4-be42-3ad8cc34ecf3.png)
![new](https://user-images.githubusercontent.com/34270884/188701800-9864264d-bccf-4907-aae3-3968a7616c50.png)

It takes much less vertical space compared to the previous one:
![old](https://user-images.githubusercontent.com/34270884/188702039-01ea1bc1-6b4c-406e-a57b-2cf9f3077c00.png)

**Reply by Pulau-Komodo** [reply]
I personally only use it on smartphone, where screen space is more precious. On large screens, there does not seem to be much of an issue either way. It does seem like the situation improved for smartphone, but only slightly.
